crates/cranelift/src/func_environ.rs

```diff
@@ -650,6 +660,33 @@ impl<'module_environment> FuncEnvironment<'module_environment> {
         self.epoch_check(builder);
     }
 
+    #[cfg(feature = "valgrind")]
+    fn check_malloc_exit(&mut self, builder: &mut FunctionBuilder, retvals: &[Value]) {
```

Comment:
Let's call this `hook_malloc_exit` I think -- it's a little clearer (we're "hooking" or inserting a hook, rather than checking right now at compile time). Likewise below for `check_free_exit`...